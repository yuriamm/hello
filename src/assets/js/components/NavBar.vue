<template>
  <div>
    <div id="nav">
      <a href="/home">Home</a> |
      <span v-if="isLoggedIn">
        <a @click="logout">Logout</a>
      </span>
      <span v-else>
        <a href="/signup">Register</a> |
        <a href="/login">Login</a>
      </span>
    </div>
  </div>
</template>

<script>
import request from "../api/request";
export default {
  name: "navbar",
  props: {
    isLoggedIn: {
      type: Boolean,
      default: false
    }
  },
  methods: {
    async logout() {
      await request("/api/logout", {}, "delete");
      sessionStorage.removeItem("user_id");
      window.location.href = "/login";
    }
  }
};
</script>
