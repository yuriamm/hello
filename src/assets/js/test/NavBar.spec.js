import { shallowMount } from "@vue/test-utils";
import NavBar from "../components/NavBar.vue";

global.window = Object.create(window);
const url = "http://localhost:4000";
Object.defineProperty(window, "location", {
  value: {
    href: url,
  },
});

const wrapper = shallowMount(NavBar);

describe("NavBar.vue", () => {
  it("snapshot macthes", async () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("snapshot macthes when logged in", async () => {
    await wrapper.setProps({ isLoggedIn: true });

    expect(wrapper.element).toMatchSnapshot();
  });

  it("to have href attribute of home", async () => {
    await wrapper.setProps({ isLoggedIn: false });
    const link = wrapper.findAll("a").at(0);

    expect(link.attributes("href")).toBe("/home");
  });

  it("to have href attribute of register", async () => {
    await wrapper.setProps({ isLoggedIn: false });
    const link = wrapper.findAll("a").at(1);

    expect(link.attributes("href")).toBe("/signup");
  });

  it("to have href attribute of login", async () => {
    await wrapper.setProps({ isLoggedIn: false });
    const link = wrapper.findAll("a").at(2);

    expect(link.attributes("href")).toBe("/login");
  });

  it("to have href attribute of logout when is logged in", async () => {
    await wrapper.setProps({ isLoggedIn: true });
    const link = wrapper.findAll("a").at(1);
    await link.trigger("click");

    expect(window.location.href).toEqual("/login");
  });
});
