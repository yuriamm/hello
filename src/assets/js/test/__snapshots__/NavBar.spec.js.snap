// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar.vue matches the snapshot 1`] = `
<div>
  <div
    id="nav"
  >
    <a
      href="/home"
    >
      Home
    </a>
     |
    
    <span>
      <a
        href="/signup"
      >
        Register
      </a>
       |
      
      <a
        href="/login"
      >
        Login
      </a>
    </span>
  </div>
</div>
`;

exports[`NavBar.vue matches the snapshot when is logged in 1`] = `
<div>
  <div
    id="nav"
  >
    <a
      href="/home"
    >
      Home
    </a>
     |
    
    <span>
      <a>
        Logout
      </a>
    </span>
  </div>
</div>
`;
